### DISCUSSION TITLE
How to force pyright to emit color escaping sequencing in a no-tty invocation

### DISCUSSION THREAD

**Discussion by vors** [microsoft/pyright#5319] [General]
When I run pyright locally on my machine I see colored output which is great.
When I run it on our remote build / test cluster (bazel RBE) I don't get any colors from it.
I assume there is some logic to avoid emitting them when pyright detects that there is no tty attached?
Is there a way to force the color emission?

**Comment by erictraut** [comment]
Pyright doesn't do anything differently. It's possible that the library it uses for emitting color commands ([chalk](https://github.com/chalk/chalk)) does. Looking at its documentation, it appears that it does perform some form of "[color support detection](https://github.com/chalk/chalk#supportscolor)". Pyright doesn't offer a way to override this detection.

**Comment by vors** [comment]
Amazing, thank you so much for the reference. Env var `FORCE_COLOR=3` does the trick.

**Reply by erictraut** [reply]
Ah, good to know!
